src/cryptography/hazmat/asn1/asn1.py

```diff
@@ -70,6 +93,15 @@ def _normalize_field_type(
                 union_args[0] if union_args[1] is type(None) else union_args[1]
             )
             annotated_type = _normalize_field_type(optional_type, field_name)
+
+            if (
+                annotation.default is not None
+                or annotated_type.annotation.default is not None
```

Comment:
Why do we need to check both of these?

Reply:
Ooof, that second one seems fairly perplexing to me. Is there a strong argument that we should support it?

Reply:
not really. I changed it so that we raise if we find annotations in the inner type of the Option, and also added a test for it.